build/ci-pipeline.yml

```diff
@@ -180,58 +340,56 @@ stages:
       subscription: $(ConnectedServiceName)
       imageTag: $(ImageTag)
 
-- stage: testStu3
-  displayName: 'Run Stu3 Tests'
+- stage: testR5
+  displayName: 'Run R5 Tests'
   dependsOn:
   - BuildUnitTests
-  - redeployStu3
-  - redeployStu3Sql
+  - redeployR5
+  - redeployR5Sql
   jobs:
   - template: ./jobs/run-tests.yml
     parameters:
-      version: Stu3
-      keyVaultName: $(DeploymentEnvironmentName)
-      appServiceName: $(DeploymentEnvironmentName)
+      version: R5
+      keyVaultName: $(DeploymentEnvironmentNameR5)
+      appServiceName: $(DeploymentEnvironmentNameR5)
 
-- stage: testR4
-  displayName: 'Run R4 Tests'
+- stage: rollbackR5
+  displayName: 'Rollback R5 CosmosDB Site'
   dependsOn:
-  - BuildUnitTests
-  - redeployR4
-  - redeployR4Sql
+  - testR5
   jobs:
-  - template: ./jobs/run-tests.yml
-    parameters:
-      version: R4
-      keyVaultName: $(DeploymentEnvironmentNameR4)
-      appServiceName: $(DeploymentEnvironmentNameR4)
+  - template: ./jobs/redeploy-webapp.yml
+    parameters: 
+      version: R5
+      webAppName: $(DeploymentEnvironmentName)
+      subscription: $(ConnectedServiceName)
+      imageTag: release
 
-- stage: testR4B
-  displayName: 'Run R4B Tests'
+- stage: rollbackR5Sql
+  displayName: 'Rollback R5 SQL Site'
   dependsOn:
-  - BuildUnitTests
-  - redeployR4B
-  - redeployR4BSql
+  - testR5
   jobs:
-  - template: ./jobs/run-tests.yml
-    parameters:
-      version: R4B
-      keyVaultName: $(DeploymentEnvironmentNameR4B)
-      appServiceName: $(DeploymentEnvironmentNameR4B)
+  - template: ./jobs/redeploy-webapp.yml
+    parameters: 
+      version: R5
+      webAppName: $(DeploymentEnvironmentNameSql)
+      subscription: $(ConnectedServiceName)
+      imageTag: release
 
-- stage: testR5
-  displayName: 'Run R5 Tests'
+- stage: testR5Rollback
+  displayName: 'Run R5 Rollback Tests'
   dependsOn:
-  - BuildUnitTests
-  - redeployR5
-  - redeployR5Sql
+  - rollbackR5
+  - rollbackR5Sql
   jobs:
   - template: ./jobs/run-tests.yml
     parameters:
       version: R5
-      keyVaultName: $(DeploymentEnvironmentNameR5)
-      appServiceName: $(DeploymentEnvironmentNameR5)
+      keyVaultName: $(DeploymentEnvironmentName)
+      appServiceName: $(DeploymentEnvironmentName)
 
+# *********************** Finalize ***********************
```

Comment:
Should the docker main tag be added after the rollback tests? Or is the goal to provide artifacts before the rollback tests run?

Reply:
That makes sense.

Leaving it as is would make a different type of sense where we'd get faster availability of the artifact if needed downstream, but we would need a manual check to ensure the pipeline passed.